index.bs

```diff
@@ -3341,6 +3341,219 @@ be the same.
+
+*   If the operation accepts multiple types for multiple arguments with no coupling between types of
+    different arguments, [=union types=] can sometimes be the only viable solution.
+
+    <pre highlight="webidl">
+        typedef (long long or DOMString or CalculatableInterface) SupportedArgument;
+        interface A {
+          void add(SupportedArgument operand1, SupportedArgument operand2);
+        };
+    </pre>
+
+    For the <code class="idl">add()</code> operation above, to specify it using
+    [=overloaded|overloads=] would require
+
+    <pre highlight="webidl">
+        interface A {
+          void add(long long operand1, long long operand2);
+          void add(long long operand1, DOMString operand2);
+          void add(long long operand1, CalculatableInterface operand2);
+          void add(DOMString operand1, long long operand2);
+          void add(DOMString operand1, DOMString operand2);
+          void add(DOMString operand1, CalculatableInterface operand2);
+          void add(CalculatableInterface operand1, long long operand2);
+          void add(CalculatableInterface operand1, DOMString operand2);
+          void add(CalculatableInterface operand1, CalculatableInterface operand2);
+        };
+    </pre>
+
+    and nine times the corresponding prose!
+
+*   Specification authors are encouraged to treat missing argument and <emu-val>undefined</emu-val>
+    argument the same way in the ECMAScript language binding.
+
+    Given the following IDL fragment:
+
+    <pre highlight="webidl">
+        interface A {
+          void foo();
+          void foo(Node? arg);
+        };
+    </pre>
+
+    Using the ECMAScript language binding, calling <code>foo(undefined)</code> and
+    <code>foo(null)</code> would both run the steps corresponding to the <code
+    class="idl">foo(|arg|)</code> operation, with |arg| set to null, while <code>foo()</code> alone
+    would go to the first overload. This may be a surprising behavior for many API users. Instead,
+    it is recommended to use an optional argument, which would categorize both <code>foo()</code>
+    and <code>foo(undefined)</code> as "|arg| is not present".
+
+    <pre highlight="webidl">
+        interface A {
+          void foo(optional Node? arg);
+        };
+    </pre>
+
+    In general, optionality is best expressed using the <emu-t>optional</emu-t> keyword, and not
+    using overloads.
+
+When the case fits none of the categories above, it is up to the specification author to choose the
+style, since it is most likely that either style would sufficiently and conveniently describe the
+intended behavior. However, the definition and <a href="#es-to-union">conversion algorithms</a> of
+[=union types=] are simpler to implement and reason about than [=overload resolution
+algorithm|those=] of [=overloaded|overloads=], and can discourage implementation mistakes. Thus,
+unless any other considerations apply, [=union types=] should be the default choice.
```

Comment:
Maybe "discourage implementation mistakes" is not the real concern here, but instead "and usually result in more idiomatic APIs in the ECMAScript binding" or something.